### PR TITLE
HIVE-24879: Create new metrics about ACID metadata size

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Introduced ACID metadata related metrics

### Why are the changes needed?

Compaction observability

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/MetricsInfo.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.apache.hadoop.hive.metastore.txn;
```
Comment (pvargacl) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/MetricsInfo.java:1]: Missing apache licence

Comment (deniskuzZ) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/MetricsInfo.java:18]: added

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java [hunk 2]
```diff
@@ -3585,6 +3587,34 @@ public ShowCompactResponse showCompact(ShowCompactRequest rqst) throws MetaExcep
     }
   }
 
+  public MetricsInfo getMetricsInfo() throws MetaException {
+    Connection dbConn = null;
+    Statement stmt = null;
+    try {
```
Comment (pvargacl) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java:3593]: can you use try-with-resources?

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java [hunk 3]
```diff
@@ -347,6 +348,38 @@ public void testAgeMetricsOrder() {
...
+    long writeid = allocateWriteId(t.getDbName(), t.getTableName(), txnid);
+    Assert.assertEquals(25, writeid);
+    txnHandler.commitTxn(new CommitTxnRequest(txnid));
+
+    // The metrics will appear after the next AcidMetricsService run
+    runAcidMetricService();
+
+    Assert.assertEquals(25,
+        Metrics.getOrCreateGauge(MetricsConstants.COMPACTION_STATUS_PREFIX + "txn_to_writeid").intValue());
+    Assert.assertEquals(1,
```
Comment (pvargacl) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java:379]: could you add one more check: call the cleanTxnToWriteIdTable and see if the metric changes to 0?

Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java:376]: added

**Review by pvargacl — APPROVED** [review]

**Pull request merged**